docs/DeveloperGuide.md

```diff
@@ -1,12 +1,406 @@
 # Developer Guide
+## Contents
+- [Acknowledgements](#acknowledgements)
+- [Design & Implementation](#design--implementation)
+- [Implementation](#implementation)
+  - [List](#list)
+  - [Add](#add)
+  - [Edit](#edit)
+  - [Remove](#remove)
+  - [Search](#search)
+  - [Filter](#filter)
+  - [Alert](#alert)
+  - [History](#history)
+  - [Category](#category)
+- [Product Scope](#product-scope)
+  - [Target User Profile](#target-user-profile)
+  - [Value Proposition](#value-proposition)
+- [User Stories](#user-stories)
+- [Non-Functional Requirements](#non-functional-requirements)
+- [Glossary](#glossary)
+- [Instructions for Manual Testing](#instructions-for-manual-testing)
 
```

Comment:
Neat table of content. Maybe could consider adding numbering here as well, corresponding to those used in the headers so the reader can navigate easier when scrolling.

For example, this could be `2.3. Edit`
![image](https://user-images.githubusercontent.com/84615041/228703518-bf5de21f-813d-499f-9ef3-86a1ee61e99a.png)
corresponding to this 
![image](https://user-images.githubusercontent.com/84615041/228703655-60719d89-0cfc-40fb-8c0d-93c394af001d.png)